src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/ProviderBase/DbConnectionPool.cs

```diff
@@ -772,7 +772,14 @@ private DbConnectionInternal CreateObject(DbConnection owningObject, DbConnectio
 
                 CheckPoolBlockingPeriod(e);
 
+                // Close associated Parser if connection already established.
+                if (newObj != null && newObj.IsConnectionAlive())
```

Comment:
Nit, this can be `newObj?.IsConnectionAlive()`

Reply:
I tried that, got an error "Cannot cast `bool?` to `bool`"

Reply:
`if (newObj?.IsConnectionAlive() == true)` could solve the cast error message.